build_projects/dotnet-cli-build/PrepareTargets.cs

```diff
@@ -145,17 +140,94 @@ public static BuildTargetResult ExpectedBuildArtifacts(BuildTargetContext c)
-            AddInstallerArtifactToContext(c, "dotnet-host", "SharedHost", hostVersion);
-            AddInstallerArtifactToContext(c, "dotnet-sharedframework", "SharedFramework", sharedFrameworkVersion);
             AddInstallerArtifactToContext(c, "dotnet-dev", "CombinedFrameworkSDKHost", cliVersion);
-            AddInstallerArtifactToContext(c, "dotnet", "CombinedFrameworkHost", sharedFrameworkVersion);
             AddInstallerArtifactToContext(c, "dotnet-sharedframework-sdk", "CombinedFrameworkSDK", cliVersion);
             AddInstallerArtifactToContext(c, "dotnet-sdk-debug", "SdkSymbols", cliVersion);
 
+            //Downloaded Installers + Archives
+            AddInstallerArtifactToContext(c, "dotnet-host", "SharedHost", hostVersion);
+            AddInstallerArtifactToContext(c, "dotnet-sharedframework", "SharedFramework", sharedFrameworkVersion);
+            AddInstallerArtifactToContext(c, "dotnet", "CombinedFrameworkHost", sharedFrameworkVersion);
+
+            return c.Success();
+        }
+
+        [Target(
+            nameof(ExpectedBuildArtifacts), 
+            nameof(DownloadHostAndSharedFxArchives), 
+            nameof(DownloadHostAndSharedFxInstallers))]
+        public static BuildTargetResult DownloadHostAndSharedFxArtifacts(BuildTargetContext c) => c.Success();
+
+        [Target]
+        public static BuildTargetResult DownloadHostAndSharedFxArchives(BuildTargetContext c)
+        {
+            Mkdirp(Dirs.Packages);
+
+            var sharedFrameworkVersion = DependencyVersions.SharedFrameworkVersion;
+            var sharedFrameworkChannel = DependencyVersions.SharedFrameworkChannel;
+
+            var combinedSharedHostAndFrameworkArchiveFile = c.BuildContext.Get<string>("CombinedFrameworkHostCompressedFile");
+
+            AzurePublisher.DownloadFile(
+                AzurePublisher.CalculateArchiveBlob(
+                    combinedSharedHostAndFrameworkArchiveFile, 
+                    sharedFrameworkChannel, 
+                    sharedFrameworkVersion), 
+                combinedSharedHostAndFrameworkArchiveFile);
+
+            // Unpack the combined archive to shared framework publish directory
+            Rmdir(Dirs.SharedFrameworkPublish);
+            Mkdirp(Dirs.SharedFrameworkPublish);
+            if(CurrentPlatform.IsWindows)
+            {
+                ZipFile.ExtractToDirectory(combinedSharedHostAndFrameworkArchiveFile, Dirs.SharedFrameworkPublish);
+            }
+            else
+            {
+                Exec("tar", "xf", combinedSharedHostAndFrameworkArchiveFile, "-C", Dirs.SharedFrameworkPublish);
+            }
+
+            return c.Success();
+        }
+
+        [Target]
+        [BuildPlatforms(BuildPlatform.Windows, BuildPlatform.OSX, BuildPlatform.Ubuntu)]
+        public static BuildTargetResult DownloadHostAndSharedFxInstallers(BuildTargetContext c)
+        {
+            Mkdirp(Dirs.Packages);
```

Comment:
This is implicit in the sharedFrameworkInstallerFile Path below

I guess I should mkdirp on those.

---
In reply to: [64623409](https://github.com/dotnet/cli/pull/3230#discussion_r64623409) [](ancestors = 64623409)